opm/input/eclipse/Schedule/UDQ/UDQActive.hpp

```diff
@@ -114,40 +208,99 @@ class UDQActive {
+        const std::string& wg_name() const
+        {
+            return this->wgname;
+        }
+
+        /// Number of times this UDA is mentioned in this particular
+        /// combination of constraint keyword and item/limit.  Effectively,
+        /// how many wells/groups use this UDA for the same purpose.
+        std::size_t use_count{};
+
     private:
-        // The wgname is need in the update process, but it should
-        // not be exported out.
-        std::string wgname;
+        /// Name of well/group affected by this UDA.
+        ///
+        /// Misleading if the use count is greater than one.
+        std::string wgname{};
     };
 
-    class InputRecord {
+    /// Internalised representation of a UDA from the input file
+    class InputRecord
+    {
     public:
-        InputRecord() :
-            input_index(0),
-            control(UDAControl::WCONPROD_ORAT)
+        /// Default constructor.
+        ///
+        /// Creates an invalid input record that is mostly useful as a
+        /// target for a deserialisation operation.
+        InputRecord()
+            : input_index { 0 }
+            , control     { UDAControl::WCONPROD_ORAT }
         {}
 
-        InputRecord(std::size_t input_index_arg, const std::string& udq_arg, const std::string& wgname_arg, UDAControl control_arg) :
-            input_index(input_index_arg),
-            udq(udq_arg),
-            wgname(wgname_arg),
-            control(control_arg)
+        /// Constructor.
+        ///
+        /// \param[in] input_index_arg Zero-based index in order of
+        /// appearance of the UDQ used for this UDA.  Needed for restart
+        /// file output purposes.
+        ///
+        /// \param[in] udq_arg Name of UDQ used for this UDA.
+        ///
+        /// \param[in] wgname_arg Name of well/group affected by this UDA.
+        ///
+        /// \param[in] control_arg Keyword and item/limit for which this UDA
+        /// supplies the numeric value.
+        InputRecord(const std::size_t  input_index_arg,
+                    const std::string& udq_arg,
+                    const std::string& wgname_arg,
+                    const UDAControl   control_arg)
+            : input_index { input_index_arg }
+            , udq         { udq_arg }
+            , wgname      { wgname_arg }
+            , control     { control_arg }
         {}
 
-        bool operator==(const InputRecord& other) const {
-            return this->input_index == other.input_index &&
-                   this->udq == other.udq &&
-                   this->wgname == other.wgname &&
-                   this->control == other.control;
+        /// Equality predicate.
+        ///
+        /// \param[in] other Object against which \code *this \endcode will be
+        /// tested for equality.
+        ///
+        /// \return Whether or not \code *this \endcode is the same as \p
+        /// other.
+        bool operator==(const InputRecord& other) const
```

Comment:
maybe move to tu?

Reply:
> maybe move to tu?

Good idea.  I've moved this and a couple of other member functions to the implementation file.